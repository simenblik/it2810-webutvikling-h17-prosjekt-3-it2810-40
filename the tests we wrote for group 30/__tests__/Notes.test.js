import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Notes from '../src/Components/Notes.js';

describe('<Notes/>, functions', () => {
	/* Takes a snapshot of the rendered component and see if it
	matches the original snapshot.
	*/
	it('should match to previous snapshots', () => {
		const component = shallow(<Notes />);
		const tree = toJson(component);
		expect(tree).toMatchSnapshot();
	});

	// Seta s state, to simulate writing of notes
	it('should set state, as if someone writes a note', () => {
		const component = shallow(<Notes />);
		component.setState({ text: 'Hello' });
		const localState = component.state('text');
		expect(localState).toBe('Hello');
	});

	/* Getting error: TypeError: Cannot read property 'toUpperCase' of undefined
	it('should clear the notes', () => {
		const component = shallow(<Notes />);
		const onChange = jest.fn();
		component.setState({ text: 'Hello' });
		component.find('textarea').simulate(onChange);
		const localState = component.state('text');
		expect(localState).toBe('');
	}); */
});
