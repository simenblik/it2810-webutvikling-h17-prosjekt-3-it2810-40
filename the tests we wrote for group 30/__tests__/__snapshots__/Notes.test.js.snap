// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notes/>, functions should match to previous snapshots 1`] = `
<div>
  <h2>
    Here you can write notes about whatever you want.
    <br />
    The content is saved automatically. No need to save : )
  </h2>
  <form
    onSubmit={[Function]}
  >
    <textarea
      id="notes"
      onChange={[Function]}
      value={null}
    />
    <input
      type="Submit"
      value="Clear notepad"
    />
  </form>
</div>
`;
